### PR TITLE
RepositoryLocal improvement

### PR DESCRIPTION
**Pull request by KrystianMarek**
- Default docker API version set to V2 in order to reflect the current
  Artifacotry default
- maxUniqueTags exposed to the library user. Defaults to 0, as in
  Artifactory. Allows user to implment cleanup policy on new and
  existing registries/repositories.

### CODE REVIEW

File: dohq_artifactory/admin.py [hunk 1]
```diff
@@ -705,8 +705,9 @@ def __init__(
         artifactory,
         name,
         packageType=Repository.GENERIC,
-        dockerApiVersion=Repository.V1,
+        dockerApiVersion=Repository.V2,
```
Comment (beliaev-maksim) [inline dohq_artifactory/admin.py:708]: no, changing default version may break existing code. Please revert

File: dohq_artifactory/admin.py [hunk 2]
```diff
@@ -740,6 +742,7 @@ def _create_json(self):
             "propertySets": [],
             "archiveBrowsingEnabled": self.archiveBrowsingEnabled,
             "yumRootDepth": 0,
+            "maxUniqueTags": self.maxUniqueTags,
```
Comment (beliaev-maksim) [inline dohq_artifactory/admin.py:745]: is that supported by API V1 ?

**Review by beliaev-maksim — CHANGES_REQUESTED** [review]

File: dohq_artifactory/admin.py (hunk 2 above)
Comment (allburov) [reply dohq_artifactory/admin.py:745]: V1 does not support it https://www.jfrog.com/confluence/display/JFROG/Using+Docker+V1

Comment (beliaev-maksim) [reply dohq_artifactory/admin.py:745]: @allburov , thanks for checking
then I would request handling of the property depending on the API version

**Comment by KrystianMarek** [comment]
Hi @beliaev-maksim , @allburov , thanks for the review. I made the changes, what do you think ?

File: dohq_artifactory/admin.py [hunk 3]
```diff
@@ -741,6 +743,12 @@ def _create_json(self):
             "archiveBrowsingEnabled": self.archiveBrowsingEnabled,
             "yumRootDepth": 0,
         }
+        """
+        Docker V2 API specific fields
+        """
+        if Repository.V2 == self.dockerApiVersion:
```
Comment (beliaev-maksim) [inline dohq_artifactory/admin.py:749]: would be more readable to have 
```suggestion
        if self.dockerApiVersion == Repository.V2:
```

File: dohq_artifactory/admin.py [hunk 4]
```diff
@@ -707,6 +707,7 @@ def __init__(
         packageType=Repository.GENERIC,
         dockerApiVersion=Repository.V1,
         repoLayoutRef="maven-2-default",
+        maxUniqueTags=0,
```
Comment (beliaev-maksim) [inline dohq_artifactory/admin.py:710]: @allburov 
I propose to start using Python snake case convention for new attributes
```suggestion
        max_unique_tags=0,
```

later we can think how to keep backwards compatibility for already existing attributes

**Review by beliaev-maksim — CHANGES_REQUESTED** [review]

File: dohq_artifactory/admin.py (hunk 4 above)
Comment (KrystianMarek) [reply dohq_artifactory/admin.py]: @beliaev-maksim , done

File: dohq_artifactory/admin.py [hunk 5]
```diff
@@ -715,6 +716,7 @@ def __init__(
         self.repoLayoutRef = repoLayoutRef
         self.archiveBrowsingEnabled = True
         self.dockerApiVersion = dockerApiVersion
+        self.maxUniqueTags = max_unique_tags
```
Comment (beliaev-maksim) [inline dohq_artifactory/admin.py:719]:
```suggestion
        self.max_unique_tags= max_unique_tags
```

Comment (KrystianMarek) [reply dohq_artifactory/admin.py]: @beliaev-maksim , done :)

Comment (beliaev-maksim) [reply dohq_artifactory/admin.py]: excellent, thanks!

**Comment by KrystianMarek** [comment]
Hi, @beliaev-maksim are te changes ok ?

**Comment by beliaev-maksim** [comment]
@KrystianMarek 
yes, all is fine, thanks

@allburov ,
if you are fine that we use more python arguments (snake_case) from now, then I am going to merge it

**Comment by allburov** [comment]
>if you are fine that we use more python arguments (snake_case) from now, then I am going to merge it

Yep, It's unavoidable, I guess :( Let's merge it!

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**